### ISSUE TITLE
linter: import/no-cycle does not detect cycles through dynamic import(); `allowUnsafeDynamicCyclicDependency` is documented but has no effect

### ISSUE THREAD

**Issue by DarktheLegend** [oxc-project/oxc#25555]
## What happened?

`import/no-cycle` never reports dependency cycles in which at least one edge is a dynamic `import()`. The `allowUnsafeDynamicCyclicDependency` option ([documented](https://oxc.rs/docs/guide/usage/linter/rules/import/no-cycle.html), default `false`) has no effect on this: the config field is parsed but never consulted by the rule, and dynamic-import edges do not enter the module graph at all.

With upstream `eslint-plugin-import` / `eslint-plugin-import-x`, the same cycle **is** reported under the default `allowUnsafeDynamicCyclicDependency: false`; setting the option to `true` is what *allows* such cycles (see import-js/eslint-plugin-import#2265).

## Reproduction

oxlint **1.78.0** (clean `npx oxlint@1.78.0`), Linux, three files in an empty directory:

**a.ts**

```ts
export const load = () => import('./b');
export const a = 1;
```

**b.ts**

```ts
import { a } from './a';
export const b = a;
```

**.oxlintrc.json**

```json
{
  "plugins": ["import"],
  "categories": { "correctness": "off" },
  "rules": {
    "import/no-cycle": ["error", { "allowUnsafeDynamicCyclicDependency": false }]
  }
}
```

```console
$ npx oxlint@1.78.0 .
$ echo $?
0
```

No diagnostics. Also reproduces with a top-level `import('./b')` expression and with `await import('./b')` inside an async function.

**Control** — replace the dynamic edge with a static one (`a.ts` → `import { b } from './b'; export const a = b;`) and the cycle is reported from both files, so the setup itself is fine:

```console
$ npx oxlint@1.78.0 .
b.ts:1:19: error import(no-cycle): Dependency cycle detected
a.ts:1:19: error import(no-cycle): Dependency cycle detected
```

## Expected behavior

With `allowUnsafeDynamicCyclicDependency: false` (the default), the dynamic-edge cycle above should be reported — matching `eslint-plugin-import(-x)`. For comparison, `eslint-plugin-import-x` `no-cycle` with `{ "allowUnsafeDynamicCyclicDependency": false }` reports 2 errors on the same file pair.

## Analysis

In [`crates/oxc_linter/src/rules/import/no_cycle.rs` @ `oxlint_v1.78.0`](https://github.com/oxc-project/oxc/blob/oxlint_v1.78.0/crates/oxc_linter/src/rules/import/no_cycle.rs) (`main` is byte-identical as of 2026-08-12):

- `allow_unsafe_dynamic_cyclic_dependency` is declared ([L77](https://github.com/oxc-project/oxc/blob/oxlint_v1.78.0/crates/oxc_linter/src/rules/import/no_cycle.rs#L77)) and defaulted ([L86](https://github.com/oxc-project/oxc/blob/oxlint_v1.78.0/crates/oxc_linter/src/rules/import/no_cycle.rs#L86)), but never read anywhere in `run_once`/the graph traversal.
- The upstream *fail* cases for dynamic-import cycles are commented out in the test suite with `// TODO: dynamic import` ([L491–L501](https://github.com/oxc-project/oxc/blob/oxlint_v1.78.0/crates/oxc_linter/src/rules/import/no_cycle.rs#L491-L501)). The *pass* cases with `allowUnsafeDynamicCyclicDependency: true` pass trivially, since dynamic edges never enter the graph in the first place.

The option surfaced in the public docs via a docs/schema-only change (#15136), which can mislead users into believing dynamic-import cycles are covered (that is how we discovered this — we migrated a hybrid oxlint+ESLint setup and had to keep `import-x/no-cycle` on the ESLint side solely because of this gap).

## Suggested resolution

Either:

1. Include dynamic `import()` edges in the module graph and honor `allowUnsafeDynamicCyclicDependency` (report by default, allow when `true`), matching `eslint-plugin-import(-x)`; or
2. Until then, mark the option as "not implemented yet" in the docs (or drop it from the JSON schema) so the gap is explicit.
